integration/experiment/sst_evaluation/gen_plot_time.py

```diff
@@ -0,0 +1,504 @@
+                                for c,time in data['Application Totals'].items()
+                                if c.startswith('TIME_HINT_NETWORK@core-')]
+            app_non_network_time_by_core = [time - net_time
+                for time, net_time
+                in zip(app_time_by_core, app_network_time_by_core)
+                if time > 0]
+
+            non_net_time_core_total = sum(app_non_network_time_by_core)
+            non_net_time_core_count = len(app_non_network_time_by_core)
+            non_net_time_core_max = max(app_non_network_time_by_core)
+
+            imbalance = (
+                (non_net_time_core_max - non_net_time_core_total / non_net_time_core_count)
+                * non_net_time_core_count
+                / (non_net_time_core_max * (non_net_time_core_count - 1))
+            )
+
+            app_name = report['Profile'][:report['Profile'].index('_' + report['Agent'])]
+            trial = int(report['Profile'].rsplit('_', 1)[-1])
+
+            runs.append({
+                'Profile': report['Profile'],
+                'Application': app_name,
+                'Agent': agent,
+                'Variant': variant,
+                'trial': trial,
+                'Time to Solution': time_to_solution,
+                'CPU Package Energy': package_energy_to_solution,
+                'DRAM Energy': dram_energy_to_solution,
+                'Energy to Solution': package_energy_to_solution + dram_energy_to_solution,
+                'Imbalance': imbalance,
+            })
+
+            for host, data in report['Hosts'].items():
+                trace_path = report_path.parent.joinpath(report_path.stem + f'.trace-{host}')
+                for field,value in data['Application Totals'].items():
+                    if field.startswith('CPU_FREQUENCY_STATUS@core-') and value != 0:
+                        core = int(field.split('@core-', 1)[-1])
+                        core_frequencies[(report['Profile'], core)] = {
+                            'Profile': report['Profile'],
+                            'Application': app_name,
+                            'Agent': agent,
+                            'Variant': variant,
+                            'trial': trial,
+                            'Frequency': value,
+                        }
+
+                sticker_frequency = (
+                    data['Application Totals']['frequency (Hz)']
+                    / data['Application Totals']['frequency (%)']
+                    * 100) // 1e8 * 1e8
+
+                trace_df = pd.read_csv(trace_path, sep='|', comment='#', na_values='NAN')
+                trace_df['timediff'] = trace_df['TIME'].diff()
+                core_count = len([c for c in trace_df.columns if c.startswith('MSR::APERF:ACNT-core')])
+                for core in range(core_count):
+                    is_networking = (trace_df[f'REGION_HINT-core-{core}'] == '0x00000004')
```

Comment:
Comment on where the 0x4 came from.  We really need a programmatic way to get at the hints.